server/aesdsocket.c

```diff
@@ -0,0 +1,472 @@
+		}
+	}
+
+	
+	
+	
+	
+
+	while(1)	
+	{	
+		incoming_data = (char *)malloc(BYTES*sizeof(char));
+		if(incoming_data == NULL)
+		{
+			printf("malloc failed\n\r");
+			return -1;
+		}
+
+		memset(incoming_data, 0, BYTES);
+	
+/******************************************************
+
+	listen()
+	
+******************************************************/
+
+		ret_val = listen(socket_fd, 10); /* backlog value might cause problems */
+		if(ret_val == -1)
+		{
+			errnum = errno;
+			syslog(LOG_ERR, "listen() returned error. The error was %s\n", strerror(errnum));
+			
+			printf("Error: listen() returned error %d\n\r", errnum);
+			
+			free(incoming_data);
+			return -1;
+		}
+		
+		
+/******************************************************
+
+	accept()
+	
+******************************************************/
+		addr_size = sizeof servinfo_connectingaddr;
+		socket_fd_connected = accept(socket_fd, (struct sockaddr *)&servinfo_connectingaddr, &addr_size);
+		if(socket_fd_connected == -1)
+		{
+			errnum = errno;
+			syslog(LOG_ERR, "accept() returned error. The error was %s\n", strerror(errnum));
+			
+			printf("Error: accept() returned error %d\n\r", errnum);
+			
+			//free(incoming_data);
+			return -1;
+		}
+		
+		char ip_addr[INET_ADDRSTRLEN];
+		
+		//TODO: print the IP address of client
+		struct sockaddr_in *addr = (struct sockaddr_in *)&servinfo_connectingaddr;
+		
+		inet_ntop(AF_INET, &(addr->sin_addr), ip_addr, INET_ADDRSTRLEN);
+		syslog(LOG_DEBUG, "Accepted Connection from %s\n\r", ip_addr);
+		printf("Accepting connection from %s\n\r", ip_addr);
+		
+/******************************************************
+
+	recv() bytes and store them in a static buffer 
+	temporarily before realloc() it
+	
+******************************************************/		
+
+		
+		ssize_t msg_length = 0;
+
+		int num_bytes = 0;
+		 
+		//char* outgoing_data = NULL;
+		uint8_t completed = 0;
+
+		while(!completed)
+		{
+			memset(&static_buff[0], 0, BYTES);
+			
+			msg_length = recv(socket_fd_connected, &static_buff[0], BYTES, 0); // MSG_WAITALL
+			if(msg_length <0)
+			{
+				errnum = errno;
+				syslog(LOG_ERR, "recv() returned error. The error was %s\n", strerror(errnum));
+				printf("No more data is available to read. recv() returned error %d\n\r", errnum);
+				
+				free(incoming_data);
+				return -1;
+			}
+			else if(msg_length == 0)
+			{
+			//TODO:
+				printf("No more messages\n\r");
+				completed = 1;
+				break;
+			}
+			
+			int i =0;
+			for(i = 0; i<BYTES; i++)
+			{
+				if(static_buff[i] == 10)
+				{
+					//num_bytes = i+1;
+					//++i;
+					completed = 1;
+					break;
+				}
+				
+			}
+			//printf("sB %s ", &static_buff[0]);
+			if(i == BYTES)
+			{
+				num_bytes = i;
+			}
+			else
+			{
+				num_bytes = ++i;
+			}
+			
+			total_bytes += num_bytes; // (i+1)
+			printf("total_bytes: %d\n\r", total_bytes);
+			//printf("num_bytes: %d\n\r",num_bytes);
+			
+			incoming_data = (char *)realloc(incoming_data, (total_bytes));
```

Comment:
@Anshul-99  what assumption is made here about the size of RAM heap relative to the total size of all packets sent?